partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -249,41 +259,123 @@ internal object RexConverter {
-                                is Expr.Cast -> when (astKey.asType is Type.String) {
-                                    true -> rexOpPathKey(current, key)
-                                    false -> rexOpPathIndex(current, key)
-                                }
+            if (newSteps.isEmpty()) {
+                return newRoot
+            }
+
+            val fromList = mutableListOf<Rel>()
+
+            val pathNavi = newSteps.fold(newRoot) { current, step ->
+                val path = when (step) {
+                    is Expr.Path.Step.Index -> {
+                        val key = visitExprCoerce(step.key, context)
+                        val op = when (val astKey = step.key) {
+                            is Expr.Lit -> when (astKey.value) {
+                                is StringValue -> rexOpPathKey(current, key)
                                 else -> rexOpPathIndex(current, key)
                             }
-                        }
-                        is Expr.Path.Step.Symbol -> {
-                            val identifier = AstToPlan.convert(step.symbol)
-                            when (identifier.caseSensitivity) {
-                                Identifier.CaseSensitivity.SENSITIVE -> rexOpPathKey(
-                                    current,
-                                    rexString(identifier.symbol)
-                                )
-                                Identifier.CaseSensitivity.INSENSITIVE -> rexOpPathSymbol(current, identifier.symbol)
+
+                            is Expr.Cast -> when (astKey.asType is Type.String) {
+                                true -> rexOpPathKey(current, key)
+                                false -> rexOpPathIndex(current, key)
                             }
+
+                            else -> rexOpPathIndex(current, key)
                         }
-                        is Expr.Path.Step.Unpivot -> error("Unpivot path not supported yet")
-                        is Expr.Path.Step.Wildcard -> error("Wildcard path not supported yet")
+                        op
+                    }
+
+                    is Expr.Path.Step.Symbol -> {
+                        val identifier = AstToPlan.convert(step.symbol)
+                        val op = when (identifier.caseSensitivity) {
+                            Identifier.CaseSensitivity.SENSITIVE -> rexOpPathKey(
+                                current,
+                                rexString(identifier.symbol)
+                            )
+
+                            Identifier.CaseSensitivity.INSENSITIVE -> rexOpPathSymbol(current, identifier.symbol)
+                        }
+                        op
+                    }
+
+                    is Expr.Path.Step.Unpivot -> {
+                        // Unpivot produces two binding, in this context we want the value,
+                        // which always going to be the second binding
+                        val op = rexOpVarLocal(1, 1)
```

Comment:
Does it matter at all what values you put in here? Same goes for the Wildcard below.

If so, does this work when you have more than two path expressions with wildcards/unpivots? Can you add a test to `PartiQLEngineDefaultTest` for an example like: `<someList>[*].books[*].author[*].name[*].middle_initial` and maybe something similar for unpivots?

Reply:
You can check the conformance report details [here](https://github.com/partiql/partiql-lang-kotlin/actions/runs/8056990189).

I think the conformance tests should be sufficient. But I can also port the tests to `PartiQLEngineDefaultTest`

Reply:
> If so, does this work when you have more than two path expressions with wildcards/unpivots?

This doesn't work when you have more than two path expressions. See this commit: https://github.com/partiql/partiql-lang-kotlin/commit/e8e31d90d161ec6dcc1758e9c134f6f1ce6118c6.

The example above doesn't compile.

Reply:
It has to do with the `1, 1` and `1, 0` you've used for `rexOpVarLocal` for unpivots and wildcards. Those only work for the test cases you've written. Since JOINs really concatenate the LHS schema with the RHS schema, you can't always assume that the item you'll be referencing will always be at some specific index. It needs to be calculated.

Reply:
Oh I see what you mean. You are right.